src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
+        }
+
+        [Fact]
+        public void CtorTest()
+        {
+            var bw = new BackgroundWorker();
+            Assert.False(bw.IsBusy);
+            Assert.False(bw.WorkerReportsProgress);
+            Assert.False(bw.WorkerSupportsCancellation);
+            Assert.False(bw.CancellationPending);
         }
 
-        private void DoWorkWithException(object sender, DoWorkEventArgs e)
+        [Fact]
+        public void RunWorkerAsyncTwice()
+        {
+            var bw = new BackgroundWorker();
+            bw.DoWork += (sender, e) => Wait(3 * TimeoutShort);
```

Comment:
If we want to do more waiting like this in inner loop tests, we should probably significantly decrease the timeout.  As is, this one test will take a second, and we're striving to have each inner loop test suite take no more than a few seconds total (we're not there yet, obviously).